saleor/product/utils/availability.py

```diff
@@ -154,13 +149,8 @@ def get_product_price_range(
                 for variant in variants
             ]
             return MoneyRange(min(prices), max(prices))
-        price = calculate_discounted_price(
-            product=product,
-            price=product.price,
-            collections=collections,
-            discounts=discounts,
-        )
-        return MoneyRange(start=price, stop=price)
+
+        return MoneyRange(zero_money(), zero_money())
```

Comment:
Maybe it would be better to return `None` than zero price, which is a valid value. But then we'd have to change `ProductPricingInfo` GraphQL type as now the price range there is not nullable. Also, we would have to make sure that products without variants are not returned for customers in the storefront.

Reply:
We can also make the entire pricing info nullable if that makes more sense. If we return a zero price, someone will render it and people will expect a free product.

Reply:
Yes, I think we should make the `Product.pricing` field nullable and return `None` when there are no variants. Also, we should change the `Product.objects.visible_to_user()` QS to not return these products for users without `MANAGE_PRODUCTS` permission. Staff users having it should see these products (we want to keep rendering them in the dashboard).

Reply:
👍

Reply:
Since we render other unavailable-for-purchase products like out-of-stock ones, wouldn't it be expected that we do return them to the storefront if they are published? You wouldn't be able to add them to the cart and we would not show a price but it should be possible to at least see their photos and descriptions.

Reply:
I think we shouldn't do that since a product without a variant is not a valid product in our data model. This would also require changes in the storefront so I think it's easier to not return them now and if we decide to render them, implement that properly in the storefront.